src/WebJobs.Script/Description/DotNet/DotNetCompilationServiceFactory.cs

```diff
@@ -27,23 +27,39 @@ ImmutableArray<ScriptType> ICompilationServiceFactory.SupportedScriptTypes
             }
         }
 
-        private static OptimizationLevel OptimizationLevel
+        private static bool IsLocal => string.IsNullOrEmpty(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId));
```

Comment:
Can you rationalize this with `WebScriptHostManager.IsAzureEnvironment`? I'm thinking we can just add these shared settings to a class in Script so we're sharing the same logic. Can add the one for IsRemoteDebugging there as well.

These might go well on Ling's new SettingsManager, but I don't think that is in yet?